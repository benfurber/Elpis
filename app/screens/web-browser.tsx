import React, { Component } from "react";
import { StyleSheet, TouchableOpacity, View } from "react-native";
import { withMappedNavigationParams } from "react-navigation-props-mapper";

import { BackgroundContainer, Icon, WebView } from "components";
import { NavigationType } from "interfaces";
import { layout, colours } from "styles";
import { Analytics } from "utils";

interface Props {
  navigation: NavigationType;
  uri: string;
}

class WebBrowserScreen extends Component<Props> {
  componentDidMount() {
    Analytics.trackContent({
      contentType: "WebBrowser",
      contentId: "",
    });
  }

  render() {
    const { navigation, uri } = this.props;

    return (
      <BackgroundContainer>
        <View>
          <View style={styles.closeContainer}>
            <TouchableOpacity onPress={() => navigation.goBack()}>
              <Icon name="times-circle" size={30} />
            </TouchableOpacity>
          </View>
        </View>
        <View style={styles.body}>
          <WebView uri={uri} />
        </View>
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  body: {
    backgroundColor: colours.whiteTransparent,
    borderTopLeftRadius: layout.borderRadiusL,
    borderTopRightRadius: layout.borderRadiusL,
    flex: 1,
  },
  closeContainer: {
    flexDirection: "row-reverse",
    padding: layout.spacing,
  },
  ctaText: {
    paddingHorizontal: layout.spacing,
  },
  row: {
    alignItems: "baseline",
    flexDirection: "row",
  },
});

const wrappedWebBrowserScreen = withMappedNavigationParams()(WebBrowserScreen);
export {
  wrappedWebBrowserScreen as WebBrowserScreen,
  WebBrowserScreen as UnwrappedWebBrowserScreen,
};
