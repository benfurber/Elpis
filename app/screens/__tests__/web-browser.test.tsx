import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { UnwrappedWebBrowserScreen } from "../web-browser";

let navigation;
jest.mock(navigation, () => jest.fn());

describe("WebBrowser", () => {
  it("renders correctly", () => {
    const component = shallow(
      <UnwrappedWebBrowserScreen
        navigation={navigation}
        uri="https://www.bbc.co.uk/news"
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
