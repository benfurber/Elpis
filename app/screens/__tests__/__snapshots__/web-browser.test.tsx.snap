// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebBrowser renders correctly 1`] = `
<BackgroundContainer>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row-reverse",
          "padding": 10,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <customIcon
          name="times-circle"
          size={30}
        />
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.85)",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "flex": 1,
      }
    }
  >
    <WebView
      uri="https://www.bbc.co.uk/news"
    />
  </View>
</BackgroundContainer>
`;
